src/screens/collection/components/collectible-item.tsx

```diff
@@ -0,0 +1,219 @@
+import { useSelectedRpcUrlSelector } from 'src/store/settings/settings-selectors';
+import { useSelectedAccountSelector } from 'src/store/wallet/wallet-selectors';
+import { formatSize } from 'src/styles/format-size';
+import { TokenInterface } from 'src/token/interfaces/token.interface';
+import { conditionalStyle } from 'src/utils/conditional-style';
+import { isDefined } from 'src/utils/is-defined';
+import { openUrl } from 'src/utils/linking.util';
+import { formatAssetAmount } from 'src/utils/number.util';
+import { createTezosToolkit } from 'src/utils/rpc/tezos-toolkit.utils';
+import { getTransferPermissions } from 'src/utils/swap-permissions.util';
+import { mutezToTz } from 'src/utils/tezos.util';
+
+import { useCollectionStyles } from '../collection.styles';
+
+const DEFAULT_OBJKT_STORAGE_LIMIT = 350;
+
+interface Props {
+  item: TokenInterface;
+  collectionContract: string;
+  setWidth: (arg: number) => void;
+}
+
+export const CollectibleItem: FC<Props> = ({ item, collectionContract, setWidth }) => {
+  const [offer, setOffer] = useState<ObjktContractInterface | FxHashContractInterface>();
+  const selectedRpc = useSelectedRpcUrlSelector();
+  const selectedAccount = useSelectedAccountSelector();
+  const tezos = createTezosToolkit(selectedRpc);
+  const dispatch = useDispatch();
+
+  const styles = useCollectionStyles();
+
+  const { handleLayout, layoutWidth } = useLayoutSizes();
+
+  useEffect(() => setWidth(layoutWidth), [layoutWidth]);
+
+  useEffect(() => {
+    tezos.contract
+      .at<ObjktContractInterface | FxHashContractInterface>(item.highestOffer?.marketplace_contract ?? OBJKT_CONTRACT)
+      .then(setOffer);
+  }, []);
+
+  const lastPrice =
+    isDefined(item.lastPrice) && isDefined(item.lastPrice.price)
+      ? `${formatAssetAmount(mutezToTz(BigNumber(item.lastPrice?.price), item.lastPrice?.decimals))} ${
+          item.lastPrice.symbol
+        }`
+      : '---';
+
+  const highestOffer = isDefined(item.highestOffer)
+    ? `${mutezToTz(BigNumber(item.highestOffer.price), item.decimals)} ${item.symbol}`
+    : 'No offers yet';
+
+  const holders = item?.holders?.filter(holder => holder.quantity > 0).map(holder => holder.holder_address) ?? [];
+  const isHolder = holders.includes(selectedAccount.publicKeyHash);
+  const isOffersExisted = isDefined(item.highestOffer);
+  const isListed = item.lowestAsk !== null;
+  const listedByUser = item.listed ?? 0;
+  const quantityByUser =
+    item?.holders?.find(holder => holder.holder_address === selectedAccount.publicKeyHash)?.quantity ?? 0;
+
+  const isAbleToList = quantityByUser > listedByUser;
+
+  const navigateToObjktForBuy = `https://objkt.com/asset/${collectionContract}/${item.id}`;
+
+  const handlePress = async () => {
+    if (!isHolder && isListed) {
+      return openUrl(navigateToObjktForBuy);
+    }
+
+    const getTransferParams = () => {
+      if (isDefined(offer)) {
+        if ('fulfill_offer' in offer?.methods) {
```

Comment:
mb in the future we must create some enum or something like this for over the methods of objkt and hxhash